scripts/setup-dev.py

```diff
@@ -0,0 +1,146 @@
+
+        if not d:
+            d = "."
+
+        d = os.path.expanduser(d)
+
+        if not os.path.exists(d):
+            os.makedirs(d)
+
+        os.chdir(d)
+
+        command = "git clone https://github.com/votingworks/nOAuth.git"
+
+        subprocess.run([command], shell=True, check=True)
+
+        noAuthPath = d
+
+    while True:
+        if not noAuthPath:
+            d = input("Where is nOAuth located?")
+
+        d = os.path.expanduser(d)
+        if os.path.exists(d):
+            noAuthPath = d
+            break
+
+        print("That path does not exist.")
+        d = ""
+
+    os.chdir(noAuthPath)
+
+    res = input("Would you like to run nOAuth? [Y/n]")
+    if res == "n":
+        print("Please see README for running Arlo with custom nOAuth.")
+        sys.exit(0)
+
+    port = ""
+    while True:
+        res = input("What port should nOAuth be run on? [8080]")
+
+        if not res:
+            res = "8080"
+
+        try:
+            int(res)
+            port = res
+            break
+        except ValueError:
+            print('"{}" is not a valid port.'.format(res))
+
+    print("Running nOAuth on port {}".format(res))
+    command = "PORT=8080 pipenv run python app.py"
+    # pylint: disable=subprocess-popen-preexec-fn
+    child = subprocess.Popen(
+        [command],
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        preexec_fn=os.setsid,
+    )
+
+    children.append(child)
+
+    res = input("Would you like to install the Arlo dev environment? [y/N]")
+
+    if res == "y":
+        subprocess.run(["make dev-environment"], check=True)
+
+    orgname = input("What is the name of your test organization? [test]")
+
+    if not orgname:
+        orgname = "test"
+
+    os.chdir(owd)
+
+    print("Setting up test org")
+    output = subprocess.run(
+        ["pipenv run python -m scripts.create-org {}".format(orgname)],
+        shell=True,
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+
+    orgid = output.stdout.strip()
+    print(orgid)
+
+    email = input("What is the email for the audit administrator? [test@test.test]")
+
+    if not email:
+        email = "test@test.test"
+
+    subprocess.run(
+        ["pipenv run python -m scripts.create-admin {} {}".format(orgid, email)],
+        shell=True,
+        check=True,
+    )
+
+    print("Configuring arlo environment")
+    os.environ["ARLO_AUDITADMIN_AUTH0_BASE_URL"] = "http://localhost:{}".format(port)
+    os.environ["ARLO_JURISDICTIONADMIN_AUTH0_BASE_URL"] = "http://localhost:{}".format(
+        port
+    )
+
+    os.environ["ARLO_AUDITADMIN_AUTH0_CLIENT_ID"] = "test"
+    os.environ["ARLO_JURISDICTIONADMIN_AUTH0_CLIENT_ID"] = "test"
+
+    os.environ["ARLO_AUDITADMIN_AUTH0_CLIENT_SECRET"] = "secret"
+    os.environ["ARLO_JURISDICTIONADMIN_AUTH0_CLIENT_SECRET"] = "secret"
+
+    print("Running Arlo. Happy auditing!")
+    subprocess.run(["./run-dev.sh"], check=True)
```

Comment:
do you need to set up the db first by running `make resetdb`?

Reply:
I'm not sure. The install steps will do that, and I'm assuming here that if you've gotten this far you've already done it. The upside to doing it that way is that subsequent runs of the script don't erase credentials, so if you run the script twice with the same credentials you don't have to log in again if your browser saves the session. The downside is that if we monkey with the data model and don't explicitly `make resetdb`, it breaks. Thoughts?

Reply:
another option is to run the latest db migrations (if there are any). we do this on every heroku release in order to keep the db up to date while preserving the existing data: https://github.com/votingworks/arlo/blob/main/Procfile#L1

Reply:
ah, that sounds like the right answer!